instance/conn_name.go

```diff
@@ -74,3 +75,9 @@ func ParseConnName(cn string) (ConnName, error) {
 	}
 	return c, nil
 }
+
+// ConnectionNameResolver resolves  This allows an application to replace the default
+// DNSInstanceConnectionNameResolver with a custom implementation.
+type ConnectionNameResolver interface {
+	Resolve(ctx context.Context, name string) (instanceName ConnName, err error)
```

Comment:
The named return values are unnecessary. All consider adding a godoc to `Resolve` to explain what this should do?

Reply:
fixed.